File: Array/containsDuplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        unorderedMap = {}

        for num in nums:
            if(unorderedMap.get(num) == None) : unorderedMap.__setitem__(num,True)
            else : return True
        
        return False

File: Array/containsDuplicate.js
// function containsDuplicate(nums) {
//   const hash = {};

//   for (const num of nums) {
//     if (hash[num]) return true;
//     else hash[num] = true;
//   }

//   return false;
// }

function containsDuplicate(nums) {
  const hash = {};

  for (const num of nums) {
    if (hash[num]) return true;

    hash[num] = true;
  }

  return false;
}
